### ISSUE TITLE
Calc methods to obtain frame's spatial velocities.

### ISSUE THREAD

**Issue by amcastro-tri** [RobotLocomotion/drake#9963]
Similarly to `EvalBodySpatialVelocityInWorld()` we want to have `CalcFrameSpatialVelocityInWorld()`.
More generally, similarly to having `CalcRelativeTransform()`, we'd like having `CalcRelativeSpatialVelocity()`.

A draft implementation looks something like this:
```c++
auto X_BF = Frame::GetFixedPoseInBodyFrame();
auto p_BF = X_BF.translation();
auto X_WB =EvalBodyPoseInWorld(context, frame.body());
auto p_BF_W = X_WB.linear() * p_BF;
auto V_WB = EvalBodySpatialVelocityInWorld(context, frame.body());
auto V_WF = V_WB.Shift(p_BF_W);
```

per @siyuanfeng-tri request.

**Comment by amcastro-tri** [comment]
More generally we want to have (`EvalFoo()` counterparts would be nice if we cache frame's kienematics) :
- `Body::EvalPoseInWorld()`
- `Frame::CalcPoseInWorld(context)` 
- `Frame::CalcRelativePose(context, frame_M)`: would return `X_MF` for calling frame `F`.
- `Body::EvalSpatialVelocityInWorld(context)`
- `Frame::CalcSpatialVelocityInWorld(context)`
- `Frame::CalcRelativeSpatialVelocity(context, frame_M, frame_E)`: would return `V_MF_E` for calling frame` F`.

**Comment by amcastro-tri** [comment]
addressed in #10206
